File: adventofcode/day_20.js
'use strict';

const input = 36000000;

const plate = new Array(1000000);

const plateSize = plate.length;

plate[0] = plate[1] = false;
for (let i = 2; i < plateSize; ++i) {
  plate[i] = true;
}

const primes = [];

for (let i = 2; i < plateSize; ++i) {
  if (!plate[i]) {
    continue;;
  }
  primes.push(i);

  for (let j = i + i; j < plateSize; j += i) {
    plate[j] = false;
  }
}

const primesLen = primes.length;

function getFactorsSum (n) {
  const originN = n;
  let total = 1;

  primes.some(p => {
    if (p > n) {
      return true;
    }
    let subTotal = 1;
    let primeProduct = p;
    while (n % p == 0) {
      n /= p;
      subTotal += primeProduct;
      primeProduct *= p;
    }
    total *= subTotal;

    return n == 1;
  });

  if (n > 1) {
    throw originN;
  }
  return total;
}

let i = 100000;
let sum = getFactorsSum(i)
let checkPoint = i;
for (; sum * 10 < input; i += 1) {
  if (i >= checkPoint) {
    console.log('check', i);
    checkPoint = i + 1000;
  }
  sum = getFactorsSum(i);
}

console.log(i);

File: adventofcode/2015/day_20.py

INPUT = 36000000

plateSize = 10000000

plate = list(range(plateSize))

plate[0] = plate[1] = False
for i in xrange(2, plateSize):
  plate[i] = True

primes = []

for i in xrange (2, plateSize):
  if not plate[i]:
    continue

  primes.append(i)

  for j in xrange(i + i, plateSize, i):
    plate[j] = False

print primes[0:10]

def getFactorsSum (n):
  originN = n
  total = 1

  for p in primes:
    if p > n:
      break

    subTotal = 1
    primeProduct = p

    while (n % p) == 0:
      n /= p
      subTotal += primeProduct
      primeProduct *= p

    total *= subTotal
    if n == 1:
      break

  if n > 1:
    raise Exception(originN)

  return total

watchPoint = 10000
for i in xrange(1000, plateSize):
  if i % watchPoint == 0:
    print i

  if getFactorsSum(i) * 10 >= INPUT:
    print i
    break
